```
15:26:25 =================================== FAILURES ===================================
15:26:25 __________________________ TestOperators.test_flatten __________________________
15:26:25 [gw0] linux2 -- Python 2.7.12 /usr/bin/python2
15:26:25 
15:26:25 self = <test_operators.TestOperators testMethod=test_flatten>
15:26:25 
15:26:25     def test_flatten(self):
15:26:25         # Flatten is a special case of Reshape when the output is a 2-D tensor.
15:26:25         x = Variable(torch.randn(1, 2, 3, 4), requires_grad=True)
15:26:25 >       self.assertONNX(lambda x: x.view(x.size()[0], x.numel() // x.size()[0]), x)
15:26:25 
15:26:25 /var/lib/jenkins/workspace/test/onnx/test_operators.py:347: 
15:26:25 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
15:26:25 /var/lib/jenkins/workspace/test/onnx/test_operators.py:63: in assertONNX
15:26:25     onnx_model_pb = export_to_string(m, args, **kwargs)
15:26:25 /var/lib/jenkins/workspace/test/onnx/test_operators.py:34: in export_to_string
15:26:25     torch.onnx.export(model, inputs, f, *args, **kwargs)
15:26:25 /var/lib/jenkins/.local/lib/python2.7/site-packages/torch/onnx/__init__.py:30: in export
15:26:25     return utils.export(*args, **kwargs)
15:26:25 /var/lib/jenkins/.local/lib/python2.7/site-packages/torch/onnx/utils.py:87: in export
15:26:25     aten, export_raw_ir=export_raw_ir)
15:26:25 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
15:26:25 
15:26:25 model = FuncModule(
15:26:25   (params): ParameterList()
15:26:25 )
15:26:25 args = tensor([[[[-0.1117, -0.4966,  0.1631, -0.8817],
15:26:25           [ 0.0539,  0.6684, -...  1.4778, -0.1696, -0.9919],
15:26:25           [-1.4569,  0.2563, -0.4030,  0.4195]]]])
15:26:25 f = <_io.BytesIO object at 0x7faebefff650>, export_params = True
15:26:25 verbose = False, training = False, input_names = None, output_names = None
15:26:25 aten = False, export_raw_ir = False, export_type = 1, example_outputs = None
15:26:25 propagate = False
15:26:25 
15:26:25     def _export(model, args, f, export_params=True, verbose=False, training=False,
15:26:25                 input_names=None, output_names=None, aten=False, export_raw_ir=False,
15:26:25                 export_type=ExportTypes.PROTOBUF_FILE, example_outputs=None, propagate=False):
15:26:25         graph, params, torch_out = _model_to_graph(model, args, f, verbose,
15:26:25                                                    training, input_names,
15:26:25                                                    output_names, aten, export_raw_ir,
15:26:25                                                    example_outputs, propagate)
15:26:25     
15:26:25         # TODO: Don't allocate a in-memory string for the protobuf
15:26:25         from torch.onnx.symbolic import _onnx_opset_version
15:26:25         defer_weight_export = export_type is not ExportTypes.PROTOBUF_FILE
15:26:25         if export_params:
15:26:25 >           proto, export_map = graph.export(params, _onnx_opset_version, defer_weight_export, export_raw_ir)
15:26:25 E           RuntimeError: ONNX export failed: Couldn't export operator aten::_cast_Double
```

--- a/torch/onnx/symbolic.py
+++ b/torch/onnx/symbolic.py
@@ -605,7 +605,7 @@ def type_as(g, self, other):
         return self
     else:
         other_type_name = self.type().scalarType().lower()
-        return g.op("Cast", self, to_i=cast_pytorch_to_onnx[other_type_name])
+        return g.op("Cast", self, to_i=scalar_name_to_pytorch[other_type_name])
 
 
 # ignore clone operators that are inserted by PyTorch autograd
@@ -687,16 +687,26 @@ def _unique(g, input, sorted, return_inverse):
 # TODO: remove these once we support Type's in the JIT IR and we can once again
 # use the unified toType operator
 cast_pytorch_to_onnx = {
-    'uint8_t': torch.onnx.TensorProtoDataType.UINT8,
-    'int8_t': torch.onnx.TensorProtoDataType.INT8,
-    'double': torch.onnx.TensorProtoDataType.DOUBLE,
-    'float': torch.onnx.TensorProtoDataType.FLOAT,
+    'Byte': torch.onnx.TensorProtoDataType.UINT8,
+    'Char': torch.onnx.TensorProtoDataType.INT8,
+    'Double': torch.onnx.TensorProtoDataType.DOUBLE,
+    'Float': torch.onnx.TensorProtoDataType.FLOAT,
     'Half': torch.onnx.TensorProtoDataType.FLOAT16,
-    'int': torch.onnx.TensorProtoDataType.INT32,
-    'int64_t': torch.onnx.TensorProtoDataType.INT64,
-    'int16_t': torch.onnx.TensorProtoDataType.INT16,
+    'Int': torch.onnx.TensorProtoDataType.INT32,
+    'Long': torch.onnx.TensorProtoDataType.INT64,
+    'Short': torch.onnx.TensorProtoDataType.INT16,
 }
 
+scalar_name_to_pytorch = {
+    'uint8_t': 'Byte',
+    'int8_t': 'Char',
+    'double': 'Double',
+    'float': 'Float',
+    'half': 'Half',
+    'int': 'Int',
+    'int64_t': 'Long',
+    'int16_t': 'Short',
+}
 
 def _cast_func_template(to_i, g, input, non_blocking):
     return g.op("Cast", input, to_i=to_i)

--- a/torch/onnx/symbolic.py
+++ b/torch/onnx/symbolic.py
@@ -605,7 +605,7 @@ def type_as(g, self, other):
         return self
     else:
         other_type_name = self.type().scalarType().lower()
-        return g.op("Cast", self, to_i=scalar_name_to_pytorch[other_type_name])
+        return g.op("Cast", self, to_i=cast_pytorch_to_onnx[scalar_name_to_pytorch[other_type_name]])
 
 
 # ignore clone operators that are inserted by PyTorch autograd

--- a/torch/onnx/symbolic.py
+++ b/torch/onnx/symbolic.py
@@ -708,6 +708,7 @@ scalar_name_to_pytorch = {
     'int16_t': 'Short',
 }
 
+
 def _cast_func_template(to_i, g, input, non_blocking):
     return g.op("Cast", input, to_i=to_i)
 

